implementation/discovery/api/src/main/java/org/inaetics/dronessimulator/discovery/api/Discoverer.java

```diff
@@ -28,6 +27,11 @@
     void unregister(Instance instance) throws IOException;
 
     /**
+     * Updates the properties associated with an instance. The instance must exist!
+     */
+     Instance updateProperties(Instance instance, Map<String, String> properties) throws IOException;
```

Comment:
At first glance this interface seems a bit weird. I personally would try to make the use case clear in the method documentation.